torch/_torch_docs.py

```diff
@@ -1242,6 +1242,46 @@ def merge_dicts(*dicts):
     tensor([ True, False, False])
 """.format(**common_args))
 
+add_docstr(torch.bitwise_left_shift,
+           r"""
+bitwise_left_shift(input, other) -> Tensor
+
+Computes the bitwise shift of :attr:`input` to the left by :attr:`other` bits.
+The input tensor must be of integral types.
+
+Args:
+    input: the first input tensor
+    other: the second input tensor
+
+Keyword args:
+    {out}
+
+Example:
+
+    >>> torch.bitwise_left_shift(torch.tensor([-1, -2, 3], dtype=torch.int8), torch.tensor([1, 0, 3], dtype=torch.int8))
+    tensor([-2, -2, 24], dtype=torch.int8)
+""".format(**common_args))
+
+add_docstr(torch.bitwise_right_shift,
+           r"""
+bitwise_right_shift(input, other) -> Tensor
```

Comment:
For a left shift it doesn't matter whether these are logical or arithmetic shifts (https://en.wikipedia.org/wiki/Arithmetic_shift#Equivalence_of_arithmetic_and_logical_left_shifts_and_multiplication), but for a right shift it does matter.

Is this a logical or an arithmetic shift? We should be clear to document it. Note I think the Python Array API underspecifies its "bitwise_right_shift", actually. cc @rgommers

Reply:
The `<<`, `>>` operators, numpy and pytorch are all consistent with each it looks like:
```
>>> 10 << 1
20
>>> 10 >> 1
5
>>> -10 >> 1
-5
>>> -11 >> 1
-6

>>> t = torch.Tensor([10])
>>> t
tensor([10.])
>>> t << 1
tensor([20.])
>>> t >> 1
tensor([5.])
>>> t = torch.Tensor([-10])
>>> t << 1
tensor([-20.])
>>> t >> 1
tensor([-5.])

>>> def f(x, shift=1):
...     x2 = np.right_shift(x, shift)
...     print(f'Original ({x}):  {np.binary_repr(x)}')
...     print(f'Shifted ({x2}): {np.binary_repr(x2)}')
... 
... 
>>> f(10)
Original (10):  1010
Shifted (5): 101
>>> f(-10)
Original (-10):  -1010
Shifted (-5): -101
```
IIRC this is arithmetic shifting for negative integers. I'll open an issue to double check and clarify this.

Reply:
This is resolved in the specification now, it says to use arithmetic shift.